### PR TITLE
Add critical status to the IndexTable

### LINKED ISSUE #9873: Add the ability to have red background rows on the Polaris IndexTable
## Issue summary
For the suspended products, we need to have a critical background in the IndexTable. This PR is to add the critical status and background.

Screenshot 2023-08-02 at 9 29 39 AM

## Steps to reproduce the problem

1. First step
1. Second step

## Reduced test case

The best way to get your bug fixed is to provide a reduced test case. This [CodeSandbox template](https://codesandbox.io/s/q82mlq0m26) is a great starting point.

## Specifications

- Are you using the React components? (Y/N):
- Polaris version number:
- Browser:
- Device:
- Operating System:

Or run `npx envinfo --system --binaries --browsers --npmPackages react,react-dom,@shopify/polaris` to provide specifications on your environment including version numbers, browser, device, and operating system.

Paste the results here:

```bash

```

### PR DESCRIPTION
**Pull request by fatimasajadi**
### WHY are these changes introduced?
For suspended products, we need to have critical status added to the IndexTable. This PR is to add that status.
Fixes https://github.com/Shopify/polaris/issues/9873

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

[Storybook Link](http://240.24.59.26:6006/?path=/story/all-components-indextable--with-row-status)

Screenshot 2023-08-02 at 9 37 50 AM

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

File: .changeset/quiet-cycles-matter.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@shopify/polaris': major
+---
+
+Add a new critical status to the IndexTable
```
Comment (kyledurand) [inline .changeset/quiet-cycles-matter.md]:
```suggestion
```

🔥

File: .changeset/stupid-boats-invite.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@shopify/polaris': minor
+---
+
+add a critical status to the IndexTable
```
Comment (kyledurand) [inline .changeset/stupid-boats-invite.md]:
```suggestion
---
'@shopify/polaris': minor
---

Added a critical status to the IndexTable
```

**Review by kyledurand — COMMENTED** [review]
Code looks good, I think @sarahill was looking into this recently so I pinged her for review

**Comment by fatimasajadi** [comment]
> Code looks good, I think @sarahill was looking into this recently so I pinged her for review

Thank you @kyledurand for the review. I have a qq, The `Major version in changeset/check-changesets` check is failing. Initially, I added my changes as a major change (when I ran the run changeset command), but I got it fixed afterwards. Still the CI check is failing. Do you have any idea how I can fix that?

File: .changeset/quiet-cycles-matter.md (hunk 1 above)
Comment (kyledurand) [reply .changeset/quiet-cycles-matter.md]: You'll have to delete this file

File: polaris-react/src/components/IndexTable/IndexTable.scss [hunk 3]
```diff
@@ -201,6 +201,18 @@ $loading-panel-height: 53px;
...
   }
 
+  &.statusCritical {
+    &,
+    .TableCell-first,
+    .TableCell-first + .TableCell {
+      background-color: var(--p-color-bg-critical);
+
+      #{$se23} & {
+        background-color: var(--p-color-bg-critical);
```
Comment (sarahill) [inline polaris-react/src/components/IndexTable/IndexTable.scss]:
```suggestion
        background-color: var(--p-color-bg-critical-subdued);
```

File: polaris-react/src/components/IndexTable/IndexTable.scss [hunk 4]
```diff
@@ -201,6 +201,18 @@ $loading-panel-height: 53px;
     }
   }
 
+  &.statusCritical {
+    &,
+    .TableCell-first,
+    .TableCell-first + .TableCell {
+      background-color: var(--p-color-bg-critical);
```
Comment (sarahill) [inline polaris-react/src/components/IndexTable/IndexTable.scss]:
```suggestion
      background-color: var(--p-color-bg-critical-subdued);
```

File: polaris-react/src/components/IndexTable/IndexTable.scss [hunk 5]
```diff
@@ -687,6 +699,12 @@ $loading-panel-height: 53px;
     }
   }
 
+  .statusCritical {
+    .TableCell:first-child {
+      background-color: var(--p-color-bg-critical);
```
Comment (sarahill) [inline polaris-react/src/components/IndexTable/IndexTable.scss]:
```suggestion
      background-color: var(--p-color-bg-critical-subdued);
```

File: polaris-react/src/components/IndexTable/IndexTable.scss [hunk 6]
```diff
@@ -764,6 +782,15 @@ $loading-panel-height: 53px;
     }
   }
 
+  .statusCritical {
+    .TableCell:last-child {
+      background-color: var(--p-color-bg-primary-subdued-selected);
+      #{$se23} & {
+        background-color: var(--p-color-bg-critical);
```
Comment (sarahill) [inline polaris-react/src/components/IndexTable/IndexTable.scss]:
```suggestion
        background-color: var(--p-color-bg-critical-subdued);
```

**Review by sarahill — APPROVED** [review]
Left some comments to update `bg-critical` to `bg-critical-subdued` but otherwise looks good! ✅

**Review by jesstelford — DISMISSED** [review]
Looking at [the latest Storybook](https://5d559397bae39100201eedc1-pmguyzibcd.chromatic.com/?path=/story/all-components-indextable--with-row-status&globals=polarisSummerEditions2023:true), it appears the styles aren't being applied correctly in either pre or post Summer Editions

Screenshot 2023-08-03 at 10 49 41 am

Screenshot 2023-08-03 at 10 46 56 am

Screenshot 2023-08-03 at 11 10 46 am

Screenshot 2023-08-03 at 11 13 06 am


It also doesn't appear to work in Condensed mode (Rice Cooker should be `status: 'critical'`):

Screenshot 2023-08-03 at 11 04 51 am

Or "small screen" mode:

Screenshot 2023-08-03 at 11 07 18 am

I've added the status to the stories for you to see which ones are broken in this PR: https://github.com/Shopify/polaris/pull/9878

**Comment by fatimasajadi** [comment]
> Looking at [the latest Storybook](https://5d559397bae39100201eedc1-pmguyzibcd.chromatic.com/?path=/story/all-components-indextable--with-row-status&globals=polarisSummerEditions2023:true), it appears the styles aren't being applied correctly in either pre or post Summer Editions
> 
> Screenshot 2023-08-03 at 10 49 41 am Screenshot 2023-08-03 at 10 46 56 am

@jesstelford Looks like I accidentally deleted the ampersand. I added it back here: https://github.com/Shopify/polaris/pull/9872/commits/3af25031ca891ffb0ce31654510e0538b67370de

Thank you so much for catching it!

**Review by jesstelford — COMMENTED** [review]
Much better 👍 

Looks like it's now only breaking with Zebra striping. Not sure if this is something we want to support anyway.

@sarahill what're your thoughts on this?

Screenshot 2023-08-03 at 11 20 10 am
Screenshot 2023-08-03 at 11 18 59 am

**Review by yesenia-perezcruz — APPROVED** [review]

**Comment by fatimasajadi** [comment]
> Much better 👍
> 
> Looks like it's now only breaking with Zebra striping. Not sure if this is something we want to support anyway.
> 
> @sarahill what're your thoughts on this?
> 
> Screenshot 2023-08-03 at 11 20 10 am Screenshot 2023-08-03 at 11 18 59 am

@jesstelford Since Sara is on vacation, I talked to @yesenia-perezcruz about this, and she approved the PR. Thanks for the question!

File: .changeset/stupid-boats-invite.md [hunk 7]
```diff
@@ -0,0 +1,5 @@
+---
+'@shopify/polaris': minor
+---
+
+Add a critical status to the IndexTable
```
Comment (chloerice) [inline .changeset/stupid-boats-invite.md:5]:
```suggestion
Added support for critical `IndexTable.Row` `status`
```

**Review by chloerice — APPROVED** [review]
This looks good to me! Tophatted the zebra striping stories locally with row statuses added and zebra striping always takes precedence and styles don't break. Oddly, the "With zebra striping and row status" story doesn't actually set `hasZebraStriping` on the `IndexTable`, so this went unnoticed as a false positive that the two props could work together 🧐

**Pull request merged**